controller/controller.go

```diff
@@ -192,11 +192,9 @@ func (c controller) GroupNeedsUpdate(req Request) (Request, bool, error) {
 
 func (c controller) Submit(ctx context.Context, req Request) (*task.Task, error) {
 	c.Config.Logger.Debug(ctx, "controller: handling submit")
-
 	if ok, err := ValidateSubmitRequest(req); !ok {
-		return nil, maskAny(err)
+		return nil, err
```

Comment:
masking the error results in a failed test: 
```
panic: interface conversion: error is *errgo.Err, not *controller.ValidationError [recovered]
	panic: interface conversion: error is *errgo.Err, not *controller.ValidationError
```
maskAny changes the error type to *errgo.Err. In order the cast the err to *controller.ValidationError, the submit func may not change the type.